src/optimizer/pushdown/pushdown_get.cpp

```diff
@@ -0,0 +1,56 @@
+#include "duckdb/optimizer/filter_pushdown.hpp"
+#include "duckdb/planner/operator/logical_filter.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+#include "duckdb/storage/data_table.hpp"
+using namespace duckdb;
+using namespace std;
+
+unique_ptr<LogicalOperator> FilterPushdown::PushdownGet(unique_ptr<LogicalOperator> op) {
+
+    assert(op->type == LogicalOperatorType::GET);
+    auto &get = (LogicalGet &)*op;
+    if (get.tableFilters.size() > 0){
+        return op;
+    }
+    //! FIXME: We only need to skip if the index is in the column being filtered
+    if (!get.table || get.table->storage->indexes.size() > 0){
```

Comment:
Why are you skipping the pushdown if the table has indexes?

Reply:
If we push down all the filters of a table the logical filter node is kaput. In the index scan optimizer we check for filter->get to replace for an index scan. So the index would never be used. We need to figure something out for the index optimization anyway.